docs/docs/icechunk-python/storage.md

```diff
@@ -0,0 +1,222 @@
+
+    With this option, the credentials for connecting to S3 are detected automatically from your environment.
+    This is usually the best choice if you are connecting from within an AWS environment (e.g. from EC2). [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        from_env=True
+    )
+    ```
+
+=== "Provide credentials"
+
+    With this option, you provide your credentials and other details explicitly. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1',
+        access_key_id='my-access-key',
+        secret_access_key='my-secret-key',
+        # session token is optional
+        session_token='my-token',
+        endpoint_url=None,  # if using a custom endpoint
+        allow_http=False,  # allow http connections (default is False)
+    )
+    ```
+
+=== "Anonymous"
+
+    With this option, you connect to S3 anonymously (without credentials).
+    This is suitable for public data. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1,
+        anonymous=True,
+    )
+    ```
+
+=== "Refreshable Credentials"
+
+    With this option, you provide a callback function that will be called to obtain S3 credentials when needed. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    def get_credentials() -> S3StaticCredentials:
+        return icechunk.S3StaticCredentials(access_key_id="xyz", secret_access_key="abc")
+
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1',
+        get_credentials=get_credentials,
+    )
+    ```
+
+#### AWS Limitations
+
+- Icechunk is currently incompatible with [S3 Express One Zone](https://aws.amazon.com/s3/storage-classes/express-one-zone/). See [this issue](https://github.com/earth-mover/icechunk/issues/667) for more discussion.
+
+#### Tigris
+
+[Tigris](https://www.tigrisdata.com/) is available as a storage backend for Icechunk. Functionally this storage backend is the same as S3 storage, but with a different endpoint. Icechunk provides a helper function specifically for [creating Tigris storage configurations](./reference.md#icechunk.tigris_storage). There are a few things to be aware of when using Tigris:
+
+- The `endpoint_url` parameter must be set to the Tigris endpoint. This is defaulted to `https://fly.storage.tigris.dev` but may be different depending on your Tigris configuration.
+- Tigris is a globally distributed object store by default. There are currently some bugs with consistency when the store is distributed across multiple regions. For now, it is recommended to use a single region for your Tigris store.
```

Comment:
Let's soften the language here, maybe ~bugs~ caveats, or something.

For the last sentence I'd say something like:

For now, to get the all the consistency guarantees Icechunk offers, you will need to setup your Tigris bucket as restricted to a single region, and paste a screenshot of how to do it.

Reply:
done, let me know what you think

Reply:
great